src/KM_Controls.pas

```diff
@@ -465,59 +466,78 @@   TKMForm = class(TKMPanel)
+    BlockInput: Boolean; // Blocks all input into the field, but allow focus, selection and copy selected text
+
+    OnChange: TNotifyEvent;
+    constructor Create(aParent: TKMPanel; aLeft, aTop, aWidth, aHeight: Integer; aFont: TKMFont; aSelectable: Boolean = True);
+
+    property CursorPos: Integer read fCursorPos write SetCursorPos;
+    property Selectable: Boolean read fSelectable write fSelectable;
+    property SelectionStart: Integer read fSelectionStart write SetSelectionStart;
+    property SelectionEnd: Integer read fSelectionEnd write SetSelectionEnd;
+    procedure MouseDown(X,Y: Integer; Shift: TShiftState; Button: TMouseButton); override;
+    procedure MouseMove(X,Y: Integer; Shift: TShiftState); override;
+    procedure MouseUp(X,Y: Integer; Shift: TShiftState; Button: TMouseButton); override;
+    function KeyDown(Key: Word; Shift: TShiftState): Boolean; override;
+    procedure KeyPress(Key: Char); override;
+    function KeyUp(Key: Word; Shift: TShiftState): Boolean; override;
+  end;
+
+  {EditField}
+  TKMEdit = class(TKMSelectableEdit)
+  private
+    fAllowedChars: TAllowedChars;
+    procedure SetText(aText: UnicodeString);
+  protected
+    function GetMaxLength: Word; override;
+    function IsCharValid(aChar: WideChar): Boolean; override;
+    procedure ValidateText; override;
+    function KeyEventHandled(Key: Word; Shift: TShiftState): Boolean; override;
+    function GetRText: UnicodeString;
+  public
+    Masked: Boolean; //Mask entered text as *s
     MaxLen: Word;
+    ShowColors: Boolean;
     DrawOutline: Boolean;
     OutlineColor: Cardinal;
-    OnChange: TNotifyEvent;
     OnIsKeyEventHandled: TNotifyEventKeyFunc; //Invoked to check is key overrides default handle policy or not
     constructor Create(aParent: TKMPanel; aLeft, aTop, aWidth, aHeight: Integer; aFont: TKMFont; aSelectable: Boolean = True);
 
     property AllowedChars: TAllowedChars read fAllowedChars write fAllowedChars;
-    property CursorPos: Integer read fCursorPos write SetCursorPos;
-    property SelectionStart: Integer read fSelectionStart write SetSelectionStart;
-    property SelectionEnd: Integer read fSelectionEnd write SetSelectionEnd;
     property Text: UnicodeString read fText write SetText;
-    property Selectable: Boolean read fSelectable write fSelectable;
 
     function HitTest(X,Y: Integer; aIncludeDisabled: Boolean=false): Boolean; override;
-    function KeyDown(Key: Word; Shift: TShiftState): Boolean; override;
-    procedure KeyPress(Key: Char); override;
-    function KeyUp(Key: Word; Shift: TShiftState): Boolean; override;
-    procedure MouseDown(X,Y: Integer; Shift: TShiftState; Button: TMouseButton); override;
-    procedure MouseMove(X,Y: Integer; Shift: TShiftState); override;
-    procedure MouseUp(X,Y: Integer; Shift: TShiftState; Button: TMouseButton); override;
-    procedure ControlMouseDown(Sender: TObject; Shift: TShiftState);
-    procedure Focus(aFocused: Boolean);
     procedure Paint; override;

```

Comment:
All code differences above are related to extracting TKMSelectedEdit from TKMEdit. 
I do not see any way to make it easier to 'read code differences' in that situation unfortunally.